source/georef/street_network.cpp

```diff
@@ -412,28 +450,37 @@ PathFinder::find_nearest_stop_points(const navitia::time_duration& radius,
                 // if the radius is still ok with sqrt(2) factor
                 auto sp_idx = routing::SpIdx(element.first);
                 if (duration < radius && distance_to_entry_point.count(sp_idx) == 0) {
-                    result[sp_idx] = duration;
-                    distance_to_entry_point[sp_idx] = duration;
+
+                    // free radius exclusion (optional)
+                    if (free_radius > 0 && !excluded_elements.empty()) {
+                        result[sp_idx] = exclude_sp_with_free_radius_filter(element.second,
```

Comment:
`element.first` is the idx of the stop_point rigth? If so why compare coord and not sp_idx?

Reply:
Yes, element.first is an idx. The filtering method is based on idx to match, not coord.